inference_segmentation.py

```diff
@@ -225,26 +227,22 @@ def segmentation(param,
     del fp
 
     fp = np.memmap(tp_mem, dtype='float16', mode='r', shape=(tf_len, h_padded, w_padded, num_classes))
-    pred_img = np.zeros((h_padded, w_padded), dtype=np.uint8)
+    pred_heatmap = np.zeros((h_padded, w_padded, num_classes), dtype=np.float16)
```

Comment:
extra dimension and data type changes will impact memory. Is this change necessary for creating the heatmaps or can it be handled in the heatmap function?

Reply:
I'll test the memory usage difference before and after this PR, but I don't see how we can keep the information from logits (ie to build heatmap) while still thresholding `arr1 = (arr1 > threshold)` with or "argmaxing" at this stage. Do you see any alternatives?

Reply:
I don't see any alternatives but this change almost fully negates the purpose of the temporary file for memory savings

Reply:
I'd like to discuss this with you before we merge this PR.

Reply:
This is solved